### PR TITLE
Issue #10312 Remove cyclic dependencies between jetty-home and jetty-documentation.

### LINKED ISSUE #10312: Remove jetty-home-with-docs to eliminate build time cyclic dependencies
**Jetty version(s)**
Jetty-12

**How to reproduce?**
```
mvn clean install -DskipTests -Dasciidoctor.skip=false
```
console has error for all ascii doc extension trying to capture output:
```
[INFO] Working Dir: /home/olamy/dev/sources/jetty/jetty.project/target/tests/test-bases/jetty_base_15951583927940853877
[INFO] Executing: [/home/olamy/.sdkman/candidates/java/21.ea.32-open/bin/java, -Djava.io.tmpdir=/home/olamy/dev/sources/jetty/jetty.project/target/tests/test-bases/jetty_base_11481260225660893827/work, -jar, /home/olamy/dev/sources/jetty/jetty.project/jetty-home/target/jetty-home/start.jar, --add-modules=threadpool-virtual-preview,http]
[INFO] Working Dir: /home/olamy/dev/sources/jetty/jetty.project/target/tests/test-bases/jetty_base_11481260225660893827
[INFO] Error: Unable to access jarfile /home/olamy/dev/sources/jetty/jetty.project/jetty-home/target/jetty-home/start.jar
```
It means documentation is partially generated.

This is because the asciidoc extension depends on a build and exploded jetty-home module https://github.com/eclipse/jetty.project/blob/c66f9809ba3f6a1a6cc0a27c254460c22d94c71b/documentation/jetty-asciidoctor-extensions/src/main/java/org/eclipse/jetty/docs/JettyIncludeExtension.java#L94 (BUT this is not declared in the pom whereas jetty-documentation wheres it really depend on jetty-home and an exploded) It would be better to use maven-dependency-plugin and unpack the dependency rather than depend on a build output directory of another module (and not even declared as a dependency)

BUT as jetty-home depends on jetty-documentation (to include it in the home zip) this would be a cyclic dependency...

Not sure how to fix that: get back jetty-distribution? another solution?

### PR DESCRIPTION
**Pull request by olamy**
- Issue #10312 Remove cyclic dependencies between jetty-home and jetty-documentation
- cleanup jetty home build extracting jetty-documentation
- add jetty-documentation module

### CODE REVIEW

**Comment by olamy** [comment]
this displays the documentation as http://localhost:8080/jetty-documentation/10.0.16-SNAPSHOT/
the path is really weird, but I'm not sure how to fix that, as the jetty-documentation-html.zip have the version as root path

**Comment by olamy** [comment]
I wonder if we shouldn't have jetty-javadoc module as well with the full javadoc?

**Comment by joakime** [comment]
Can we just get rid of jetty-home-with-docs archives entirely?

**Comment by olamy** [comment]
> Can we just get rid of jetty-home-with-docs archives entirely?

+1

**Comment by joakime** [comment]
Proposal.

* drop jetty-home-with-docs artifacts
* drop documentation module
* don't create a javadoc module

**Review by joakime — DISMISSED** [review]

File: documentation/jetty-documentation/pom.xml [hunk 1]
```diff
@@ -138,6 +138,30 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>unpack-documentation</id>
```
Comment (sbordet) [inline documentation/jetty-documentation/pom.xml]: This should be named `unpack-jetty-home`.

File: documentation/jetty-documentation/pom.xml [hunk 2]
```diff
@@ -248,5 +272,12 @@
       <groupId>org.eclipse.jetty.websocket</groupId>
       <artifactId>websocket-jetty-server</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty</groupId>
+      <artifactId>jetty-home</artifactId>
+      <version>${project.version}</version>
+      <type>zip</type>
+      <optional>true</optional>
```
Comment (sbordet) [inline documentation/jetty-documentation/pom.xml:280]: Why optional? This is mandatory.

File: jetty-home/src/main/resources/etc/jetty-documentation.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          <Set name="war"><Property name="jetty.base"/>/lib/jetty-documentation/jetty-documentation.war</Set>
+          <Set name="extractWAR">true</Set>
+          <Set name="copyWebDir">false</Set>
+          <Set name="defaultsDescriptor"><Property name="jetty.home" default="."/>/etc/webdefault.xml</Set>
+        </New>
+      </Arg>
+    </Call>
+  </Ref>
+
+</Configure>
```
Comment (sbordet) [inline jetty-home/src/main/resources/etc/jetty-documentation.xml]: No, remove this file.

File: jetty-home/src/main/resources/modules/jetty-documentation.mod [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+[depends]
+http
+webapp
+deploy
+
+[xml]
+etc/jetty-documentation.xml
+
+[files]
+maven://org.eclipse.jetty.documentation/jetty-documentation/${jetty.version}/zip/html|lib/jetty-documentation/jetty-documentation.war
```
Comment (sbordet) [inline jetty-home/src/main/resources/modules/jetty-documentation.mod]: No, remove this file.

**Review by sbordet — CHANGES_REQUESTED** [review]

**Review by sbordet — APPROVED** [review]

**Pull request merged**